server.ts

```diff
@@ -263,8 +264,53 @@ export function app() {
 }
 
 function run() {
+  const https = require('https');
```

Comment:
Shouldn't this check only be done if the ICM_BASE_URL actually starts with https? This check will always fail if I use a local URL without TLS